ui/src/main/java/com/doordeck/sdk/common/manager/Doordeck.kt

```diff
@@ -1,446 +1,209 @@
-     */
-    internal fun storeCertificates(certificateChain: CertificateChain) {
-        datastore.saveCertificates(certificateChain)
-    }
 
-    /**
-     * get stored certificates from keychains
-     */
-    internal fun getStoredCertificateChain(): CertificateChain? {
-        return datastore.getSavedCertificates()
+            return@runAsync
+        }
     }
 
     /**
-     * Store certificates them in the keychains
+     * Returns the current authentication status of the SDK.
      */
-    internal fun storeToken(context: Context, authToken: String) {
-        datastore.saveAuthToken(context, authToken)
-    }
+    val authStatus: AuthStatus
+        get() = requireDoordeck.authStatus
 
     /**
-     * get stored certificates from keychains
-     */
-    internal fun getStoredAuthToken(context: Context): String? {
-        return datastore.getAuthToken(context)
-    }
+     * Retrieves the headless instance of the SDK.
+     *
+     * Lazily initializes if not yet done.
+     *
+     * @param context A valid context (preferably application context).
+     * @return The internal Doordeck instance.
+     */
+    fun getHeadlessInstance(context: Context): com.doordeck.multiplatform.sdk.Doordeck {
+        if (doordeck == null) {
+            doordeck = KDoordeckFactory.initialize(
+                SdkConfig.Builder()
+                    .setApplicationContext(ApplicationContext.apply { set(context) })
+                    .build()
+            )
+        }
 
-    /**
-     * Store certificates them in the keychains
-     */
-    internal fun storeLaststatus(status: AuthStatus) {
-        datastore.saveStatus(status)
+        return requireDoordeck
     }
 
     /**
-     * get stored certificates from keychains
+     * Logs out the current user and clears all local SDK data.
+     *
+     * @return A CompletableFuture that resolves once logout completes.
      */
-    internal fun getLastStatus(): AuthStatus? {
-        return datastore.getStoredStatus()
+    fun logout(): CompletableFuture<Unit> {
+        return requireDoordeck.account().logoutAsync()
     }
 
     /**
-     * Store certificates them in the keychains
+     * Shows the device verification screen.
+     *
+     * Call this to allow the user to verify their device if needed.
+     *
+     * @param context A valid context for navigation.
      */
-    internal fun storeTheme(darkMode: Boolean) {
-        datastore.saveTheme(darkMode)
+    fun showVerificationScreen(context: Context) {
+        VerifyDeviceActivity.start(context)
     }
 
     /**
-     * get stored certificates from keychains
+     * Helper.
+     * Generates and sets a key pair if none exists.
      */
-    internal fun getSavedTheme(): Boolean {
-        val darkMode = datastore.getSavedTheme()
-        if (darkMode != null) return darkMode
-        else return false
+    private fun setKeyPairIfNeeded() {
+        if (!requireDoordeck.contextManager().isKeyPairValid()) {
```

Comment:
This method will actually sign a small text and verify its signature. If you only want to check whether the SDK has a key pair, you can use `sdk.contextManager().getKeyPair() != null`, which is a lighter check.